assets/cluster-monitoring-operator/prometheus-rule.yaml

```diff
@@ -28,6 +28,37 @@ spec:
       for: 15m
       labels:
         severity: warning
+    - alert: HighlyAvailableWorkloadIncorrectlySpread
+      annotations:
+        description: Workload {{ $labels.namespace }}/{{ $labels.workload }} is incorrectly
+          spread across mutliple nodes which breaks high-availability requirements.
+          There are {{ $value }} pods on node {{ $labels.node }}, where there should
+          only be one. Since the workload is using persistent volumesm it is not fixable
+          by operators, but you can follow the guidelines provided in the runbook
+          of this alert to fix this issue.
+        runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/HighlyAvailableWorkloadIncorrectlySpread.md
+        summary: Highly-available workload is incorrectly spread across multiple nodes
+          and is not fixable because of persitent volumes.
+      expr: |
+        count by(node,workload,namespace)
+          (
+            kube_pod_info{node!=""}
+            * on(namespace,pod) group_left()
+            (
+              kube_pod_spec_volumes_persistentvolumeclaims_info
+            )
+            * on(namespace,pod) group_left(workload)
+            (
+              namespace_workload_pod:kube_pod_owner:relabel
+              * on(namespace,workload,workload_type) group_left()
+              (
+                count without(pod) (namespace_workload_pod:kube_pod_owner:relabel) > 1
+              )
+            )
+          ) > 1
+      for: 15m
```

Comment:
The `for` clause here prevents catching surges during rolling updates.

Reply:
Let's increase it to ~30m or even higher. The condition is fairly static and there is no imminent problem with having workload on one node (if there was, the alert should be critical and not warning).

Reply:
Make sense, increased to `1h` to conform with the [Alerting guidelines](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/alerting-consistency.md#warning-alerts) for warning alerts in OCP.